### PR TITLE
Commcarehq prelogin

### PR DESCRIPTION
**Pull request by biyeun**
- splits out prelogin into https://github.com/dimagi/commcarehq-prelogin (please look to this repo to review the code for the public site)
- adds a flag to settings `ENABLE_PRELOGIN_SITE`. By default the flag is `False` and the repo is not included in urls or django apps.
- the new prelogin repo lives in `corehq/apps/public`. It lives here because it is not truly a submodule (split out for licensing purposes), and is highly dependent on the rest of `corehq`, and is, basically, part of corehq.

### CODE REVIEW

File: localsettings.example.py [hunk 1]
```diff
@@ -27,6 +27,9 @@
 #Staging domains you want to replicate
 STAGING_DOMAINS = []
 
+### Public / Pre-login Site information
```
Comment (dimagimon) [inline localsettings.example.py]: E265 block comment should start with '# '

**Comment by biyeun** [comment]
@orangejenny @czue @NoahCarnahan for review. need to test a couple things on staging first.

**Comment by czue** [comment]
fyi, I'm not going to be able to review this before next week, but happy for others to look. might make sense for @dannyroberts to take a quick peek on the overall layout/approach

**Comment by NoahCarnahan** [comment]
Not sure if this is on your radar, but this todo is still in there:
https://github.com/dimagi/commcarehq-prelogin/blob/73e51894af16257c33df88013170d5591b17e861/views.py#L27

**Comment by biyeun** [comment]
@NoahCarnahan yep

**Comment by biyeun** [comment]
build passed fyi @NoahCarnahan @orangejenny @dannyroberts

**Pull request merged**